### ISSUE TITLE
Rank 30 index during import: Extremely slow query.

### ISSUE THREAD

**Issue by grzezlo** [osm-search/Nominatim#2608]
Below problem is related to code on master branch (checked out on commit cd9b0c9a201e).
After changing to 4.0.x branch, import went well.

I tried Nominatim import on strong bare-metal server with 256 GB of RAM, and on my small WSL2 Ubuntu installation. Ubuntu 20.04 in both cases.
Tried PostgreSQL 12 and 14.


On server I went from whole planet extract, through one-country, to regional (osm.pbf taking about 200 MB).

In all cases a problem is the same:
Everything goes well until indexing rank 30  starts, where I get:

```
Done 180 in 2772 @ 0.065 per second - rank 30 ETA (seconds): 143411849.50
```

After enabling long querries logging in PostGresql (log_min_duration_statement=30000), I get one query taking about 23 minutes:

```
duration: 1384521.011 ms  statement: SELECT place_id, (placex_indexing_prepare(placex)).*
	                          FROM placex WHERE place_id IN (869888, 129306, 896096, 315402, 818588, 298802, 262370, 788657, 327650, 844126, 376045, 106647, 898342, 915995, 182535, 843334, 295198, 347854, 837115, 209352, 903116, 278955, 853174, 327693, 835619, 909836, 206466, 835510, 877540, 888413, 338972, 326143, 834540, 349406, 343632, 903283, 893024, 811810, 114527, 941699, 3742493, 3742744, 3814921, 3820708, 3815403, 3814872, 3803687, 3819872, 3804404, 3803749, 3780254, 3780196, 3797445, 3780015, 3779976, 3815075, 3742811, 3779061, 3461166, 3804585, 3800621, 3800578, 3770244, 3769232, 3769272, 3793514, 3568042, 3770193, 3829889, 3830188, 3830321, 3793952, 3820597, 3794299, 3815051, 3769970, 3816075, 3791363, 3791254, 7519207, 7312050, 7369787, 7486222, 7419343, 7402948, 7314335, 7385248, 7515892, 7399955, 7441637, 7313065, 7391937, 7340157, 7447397, 7394436, 7424596, 7445405, 7461184, 7329024, 7595433)
```

The "explain analyze" for this query is as follows:

```
                                                                                                                                                                                                                                                                                                                                                                                                              QUERY PLAN                                                                                                                                                                                                                                                                                                                                                                                                               
-------------------
 Index Scan using idx_place_id on placex  (cost=0.43..724.72 rows=100 width=178) (actual time=31952.652..1835581.289 rows=100 loops=1)
   Index Cond: (place_id = ANY ('{869888,129306,896096,315402,818588,298802,262370,788657,327650,844126,376045,106647,898342,915995,182535,843334,295198,347854,837115,209352,903116,278955,853174,327693,835619,909836,206466,835510,877540,888413,338972,326143,834540,349406,343632,903283,893024,811810,114527,941699,3742493,3742744,3814921,3820708,3815403,3814872,3803687,3819872,3804404,3803749,3780254,3780196,3797445,3780015,3779976,3815075,3742811,3779061,3461166,3804585,3800621,3800578,3770244,3769232,3769272,3793514,3568042,3770193,3829889,3830188,3830321,3793952,3820597,3794299,3815051,3769970,3816075,3791363,3791254,7519207,7312050,7369787,7486222,7419343,7402948,7314335,7385248,7515892,7399955,7441637,7313065,7391937,7340157,7447397,7394436,7424596,7445405,7461184,7329024,7595433}'::bigint[]))
 Planning Time: 19.320 ms
 Execution Time: 1835581.876 ms
(4 rows)
```

**Comment by lonvia** [comment]
There is a similar report on #2605 but I can't reproduce this locally. Could you please report what custom settings you are using for PostgreSQL and in your .env file (in particular: do you use the new ICU tokenizer).

Then, could you please run `./nominatim admin --analyse-indexing --place-id <placeID>` against each place ID listed in the query you report and see if one runs particularly long. If so, please paste the output here.

**Comment by grzezlo** [comment]
Thanks @lonvia for your reply.

As I wrote:

- The same problem appeared on bare-metal server, and WSL2 on a laptop.
- Server has PostgreSQL 12, WSL2 - PostgreSQL 14 installed.
- On server I tried making changes in the postgresql.conf settings, e.g. suggested in the #1677 issue; on the Wsl2 I tried to import on default PostGresql settings.
- On server I used flatNode file, on the Wsl2 - not.
- I didn't tried the ICU tokenizer, using default one on both machines.
- On both machines, Ubuntu 20.04 was installed.
- On both instances, solution or workaround was to switch from master branch, to 4.0.x branch and recompile everything.

I'm pretty sure, that the problem would be easy to reproduce on the WSL2 instance, but at the moment hadn't free time to check it and know for sure.

Also no time at the moment to do the suggested experiment of checking place ids one by one, but i'll do it later if needed.
